src/core/tests/type_prop/search_sorted.cpp

```diff
@@ -0,0 +1,82 @@
+// Copyright (C) 2018-2024 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#include "openvino/op/search_sorted.hpp"
+
+#include "common_test_utils/type_prop.hpp"
+
+using namespace std;
+using namespace ov;
+
+#define EXPECT_THROW_SUBSTRING(SORTED, VALUES, SUBSTRING)             \
+    try {                                                             \
+        auto op = make_shared<op::v15::SearchSorted>(SORTED, VALUES); \
+        FAIL() << "Exception not thrown";                             \
+    } catch (const NodeValidationFailure& error) {                    \
+        EXPECT_THAT(error.what(), testing::HasSubstr(SUBSTRING));     \
+    } catch (...) {                                                   \
+        FAIL() << "Unexpected exception thrown";                      \
+    }
+
+static void PerformShapeTest(const PartialShape& sorted_shape,
+                             const PartialShape& values_shape,
+                             const PartialShape& expected_output_shape) {
+    auto sorted = make_shared<op::v0::Parameter>(element::i32, sorted_shape);
+    auto values = make_shared<op::v0::Parameter>(element::i32, values_shape);
+    auto search_sorted_op = make_shared<op::v15::SearchSorted>(sorted, values);
+    EXPECT_EQ(search_sorted_op->get_element_type(), element::i64);
+    EXPECT_EQ(search_sorted_op->get_output_partial_shape(0), expected_output_shape);
+}
+
+TEST(type_prop, search_sorted_shape_infer_equal_inputs) {
+    PerformShapeTest({1, 3, 6}, {1, 3, 6}, {1, 3, 6});
```

Comment:
Add test with scalar shapes ?

Reply:
By 'scalar shape', you meant 1d shape or shape describing just one value?(eg. {1}. {1,1,1,1}, etc)?

Reply:
Scalar shape is `Shape{}` has no dimensions. Maybe it should be not allowed for inference.

Reply:
If you ask for fully dynamic case, where even rank is not known, then there is already a test case for it.

Reply:
I mean static case when shape has rank 0. Only one element but without dimensions.